### PR TITLE
add `KeyAlgorithm` enum type

### PR DESCRIPTION
**Pull request by aead**
This commit adds a enum for cryptographic
algorithms.

In general, a crypto. key is specific for
one crypto. algorithm. This commit exposes
the crypto. algorithm as enum.

Further, this commit changes the key data type
such that its binary representation is more
compact by using the `KeyAlgorithm` enum
instead of a string type.

In addition, this commit adds a version
identifier to the binary and text representation
of keys that allows easier backwards-compatible
changes to the key data type.

### CODE REVIEW

File: key.go [hunk 1]
```diff
@@ -47,12 +48,117 @@ type PCP struct {
...
+func (a KeyAlgorithm) String() string {
+	switch a {
+	case KeyAlgorithmUndefined:
+		return "undefined"
+	case AES256_GCM_SHA256:
+		return "AES256-GCM_SHA256"
+	case XCHACHA20_POLY1305:
+		return "XCHACHA20-POLY1305"
+	default:
+		return "%" + strconv.Itoa(int(a))
```
Comment (vadmeste) [inline key.go:72]: I am not sure I understand why '%'

**Review by vadmeste — COMMENTED** [review]
Few comments

File: key.go (hunk 1 above)
Comment (aead) [reply key.go]: https://github.com/minio/kes/pull/312#discussion_r1071237275

**Review by vadmeste — DISMISSED** [review]
LGTM, not tested

**Pull request merged**